### PR TITLE
Update supabase python reference docs with params

### PR DESCRIPTION
**Pull request by silentworks**
## I have read the [CONTRIBUTING.md](https://github.com/supabase/supabase/blob/master/CONTRIBUTING.md) file.

YES

## What kind of change does this PR introduce?

Docs update

## What is the current behavior?

There are no parameters and their types in the docs for the functions

## What is the new behavior?

Parameters and their types are added to the docs for the functions

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: apps/docs/spec/supabase_py_v2.yml [hunk 1]
```diff
@@ -781,26 +938,140 @@ functions:
...
           [JSON](/docs/guides/database/json) columns. Postgres offers some
           [operators](/docs/guides/database/json#query-the-jsonb-data) for
           querying JSON data.
         hideCodeBlock: true
+      - id: querying-referenced-table-with-inner-join
+        name: Querying referenced table with inner join
+        code: |
+          ```python
+          response = (
+              supabase.table("cities")
```
Comment (J0) [inline apps/docs/spec/supabase_py_v2.yml:964]: New addition

File: apps/docs/spec/supabase_py_v2.yml [hunk 2]
```diff
@@ -1456,155 +3072,313 @@ functions:
...
+              .text_search(
+                  "catchphrase",
+                  "'fat' & 'cat'",
+                  options={"type": "plain", "config": "english"},
+              )
+              .execute()
+          )
           ```
-          APIResponse(data=[{'id': 1, 'name': 'Michael', 'address': {'postcode': 90210}}], count=None)
+      - id: full-normalization
```
Comment (J0) [inline apps/docs/spec/supabase_py_v2.yml:3176]: 🔥

**Review by J0 — COMMENTED** [review]
Skimmed, looks great to me

File: apps/docs/spec/supabase_py_v2.yml [hunk 3]
```diff
@@ -64,91 +155,267 @@ functions:
...
+                  {
+                    "id": "11111111-1111-1111-1111-111111111111",
+                    "user_id": "11111111-1111-1111-1111-111111111111",
+                    "identity_data": {
+                      "email": "hello@example.com",
+                      "sub": "11111111-1111-1111-1111-111111111111"
+                    },
+                    "provider": "email",
+                    "created_at": "2024-06-17T00:19:25.774522Z",
+                    "last_sign_in_at": "2024-06-17T00:19:25.774498Z",
```
Comment (J0) [inline apps/docs/spec/supabase_py_v2.yml:255]: /nit Worried about keeping this in sync with schemachanges but not too big a concern.  I think there's an `is_anonymous` response now

**Review by J0 — APPROVED** [review]

File: apps/docs/spec/supabase_py_v2.yml (hunk 3 above)
Comment (silentworks) [reply apps/docs/spec/supabase_py_v2.yml:255]: Good catch. I will add it in a new PR with further updates. I think in order to keep schema changes in sync we will just have to make sure that the Pydantic model is kept up to date and it's response is always added to the docs.

**Pull request merged**